openwisp_controller/config/tests/test_notifications.py

```diff
@@ -65,15 +66,12 @@ def test_device_registered(self):
             notification = notification_qs.first()
             self.assertIn('The existing device', notification.message)
 
+    @patch('openwisp_notifications.types.NOTIFICATION_TYPES', {})
+    @patch('openwisp_utils.admin_theme.dashboard.DASHBOARD_CHARTS', {})
```

Comment:
@pandafy I wonder why this particular line? Only spotted it now.

Reply:
Calling `app.ready()` will re-register notification types and dashboard charts which will raise `ImproperlyConfigured` exception. 

Re-registering of both notification type or dashboard chart with same name/configuration is not allowed. 

Mocking these `dict` is to prevent re-registering of notification types and dashboard chart. 

When I wrote this code earlier, I manually unregistered individual notification types to avoid re-registration. But, this looks like a cleaner way to achieve the save.

Reply:
ok thanks for the explanation